enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,455 @@
+scenarios.
+
+##### Managed Cluster Phase 2 - Add Resiliency
+
+Phase 1 has the limitation of not being able to withstand disconnects from the hub. This would prevent a user from
+relying on the data where the following events can occur: network connectivity is flaky, the managed cluster is expected
+to be disconnected from the hub (e.g. managed cluster on a ship), or the hub is offline for maintenance.
+
+To account for this, we need a queueing mechanism on each managed cluster that stores compliance events when the hub is
+not reachable. A queuing mechanism in memory on the Status Sync controller would be the simplest approach, however, this
+would not survive container restarts and could consume a large amount of memory for a small cluster. Another approach
+would be to have a message broker such as RabbitMQ that supports queues on disk, however, the resource utilization on
+small clusters would be too large and it brings with it a maintenance cost of Open Cluster Management (OCM) needing to
+maintain this deployment for the user. Another option is using SQLite as a queue, however, it would need to be on a
+persistent volume and thus, the newer [write-ahead-log](https://www.sqlite.org/wal.html) concurrency option is not an
+option due to it potentially being on network storage. The older concurrency options involve an
+[exclusive lock](https://www.sqlite.org/lockingv3.html). This would mean that future evolutions that involve concurrency
+such as that described in phase 3 would not be supported.
+
+The proposed approach is to use [embedded etcd](https://pkg.go.dev/go.etcd.io/etcd/server/v3/embed) as this is very
+light weight as shown in the implementation details section below. This supports concurrent writes and has an HTTP
+endpoint for phase 3. This might seem unconventional since etcd is a key-value store and not a queue per se, but we can
+use it in this manner.
+
+The Governance Policy Framework addon pod would run a separate goroutine that runs embedded etcd. In this phase, this
+would have the etcd GRPC endpoint limited to localhost to avoid needing authentication enabled. The Status Sync
+controller would first try to record the compliance event on the hub, however, if there is no connection, the compliance
+event would be recorded in etcd with the key format of
+`compliance-events/<timestamp from Kubernetes event>-<event-uid>`. For example,
+`compliance-events/2023-07-25T13:15:08Z-4b400c5f-3828-4138-ad4f-03de4d69655b`. The value would be in the format of the
+`POST` request for the `/api/v1/compliance-events` HTTP endpoint on the hub.
+
+The Governance Policy Framework addon pod would run a separate goroutine that would periodically poll etcd for new
+compliance events and try to record them on the hub. It would do this in order by sorting by the timestamps in the keys.
+When successful, it would delete the entry in etcd.
```

Comment:
Why not always send the events to the embedded etcd first, and then always just POST them to the hub from that? I'm thinking that sending the event to the hub would be the slow part of the operation, so waiting for that to complete (or fail) would limit the number of policies the status-sync can handle at once.

Reply:
I was thinking to limit the chance that the message is lost since we aren't guaranteed a persistent volume but we can just prevent the poller/syncer goroutine from letting the pod shutdown until all compliance events are synced on the hub when embedded etcd is not backed by a persistent volume.

I change the design to have it like this.